### PR TITLE
Use base merging for imports

### PR DESCRIPTION
**Pull request by gouttegd**
This PR switches FBbt to the “base merging” system for imported ontologies.

In the process, it adds CL, SO, NCBITaxon, and CARO as imports, because we are already using a handful of terms from these ontologies in some of our definitions.

### CODE REVIEW

**Comment by gouttegd** [comment]
@matentzn Same as for FBdv, a double-check would be appreciated.

Especially on the issue of adding CARO, CL, SO, and NCBITaxon as imports.

As for FBdv, I’ve pushed a “mock” release PR (#1616), if you want to have a look at the generated artefacts. There’s nothing obviously wrong in them AFAIK.

File: src/ontology/fbbt-odk.yaml [hunk 1]
```diff
@@ -34,11 +34,28 @@ components:
...
+        - http://purl.obolibrary.org/obo/GO_
+        - http://purl.obolibrary.org/obo/GOREL_
     - id: ro
+      use_base: TRUE
     - id: pato
-      module_type: minimal
+      use_base: TRUE
+    - id: caro
+      use_base: TRUE
+    - id: CL
```
Comment (matentzn) [inline src/ontology/fbbt-odk.yaml:53]:
```suggestion
    - id: cl
```

File: src/ontology/fbbt-odk.yaml [hunk 2]
```diff
@@ -34,11 +34,28 @@ components:
...
+      use_base: TRUE
+    - id: SO
```
Comment (matentzn) [inline src/ontology/fbbt-odk.yaml:55]:
```suggestion
    - id: so
```

**Review by matentzn — COMMENTED** [review]
Just made two comments about odd capitalisation. Did you make sure there are no dangling classes? You have very few exclusion Iris for my intuition..

File: src/ontology/fbbt-odk.yaml [hunk 3]
```diff
@@ -34,11 +34,28 @@ components:
...
+      make_base: TRUE
+    - id: ncbitaxon
+      mirror_from: http://purl.obolibrary.org/obo/ncbitaxon/subsets/taxslim.owl
```
Comment (matentzn) [inline src/ontology/fbbt-odk.yaml:58]: You should use make_base and also add the correct base iri (be careful of the capitalisation).

Comment (gouttegd) [reply src/ontology/fbbt-odk.yaml:58]: You mean something like

```
- id: ncbitaxon
  mirror_from: http://purl.obolibrary.org/obo/ncbitaxon/subsets/taxslim.owl
  make_base: TRUE
  base_iris:
    - http://purl.obolibrary.org/obo/NCBITaxon_
```

If so I am not sure I understand why declaring the base IRI would be needed here (I note there is no such declaration in CL and Uberon’ configs).

Comment (matentzn) [reply src/ontology/fbbt-odk.yaml:58]: Because robot is case sensitive with base iris and ODK makes an all uppercase assumption

Comment (gouttegd) [reply src/ontology/fbbt-odk.yaml:58]: Doh! Got it, thanks.

**Comment by gouttegd** [comment]
> Did you make sure there are no dangling classes?

No more than the ones we already had before with non-merged imports. And they are from ontologies we actually really import from (CL’ `cell`, PATO’s `quality`, SO’s `sequence feature`, GO’s `biological_process`, `molecular_function` and `cellular_component`), so their presence is actually expected.

The only extra dangling classes that resulted from the merged imports were a dozen of Uberon classes, hence the explicit exclusion.

**Review by matentzn — APPROVED** [review]
Looks great to me!

**Pull request merged**